addons/spectator/functions/fnc_compat_spectatorBI.sqf

```diff
@@ -22,6 +22,10 @@ if !(QUOTE(ADDON) in _templates) exitWith {};
 // Start our spectator
 [true] call FUNC(setSpectator);
 
+// Delete the seagull that spawns (not actually the player, a CfgNonAIVehicles object)
+// This could delete seagulls created by a wildlife module (TODO: Find a more accurate method of getting the exact seagull)
```

Comment:
Testing shows that the template is only passed the seagull with respawn type `1`, so for types `4` and `5` I still have to do a workaround. It's an improvement though.